### PR TITLE
Save/load .hexrd instrument config files

### PR DESCRIPTION
**Pull request by psavery**
These are HDF5 files that hexrd is now capable of saving/loading.

This is now the default instrument config file type, but yaml files
can still be loaded as well.

Now also, the way the files are saved/loaded internally has been changed.
Before, a dict was just written out to a yaml file. Now, an HEDMInstrument
object is created and used to write out either the yaml/hdf5 file.

Also, save/load of .hexrd directories is no longer supported, since it is
not needed anymore.

A util function was also added to extract out the config dict from an
HEDMInstrument object and perform any necessary conversions.

### CODE REVIEW

File: hexrd/ui/utils.py [hunk 1]
```diff
@@ -245,3 +245,39 @@ def exclusions_off(plane_data):
...
+    finally:
+        # Restore the panel buffers
+        for key, panel in instr.detectors.items():
+            panel.panel_buffer = panel_buffers[key]
+
+    # Set the panel buffers on the detectors
+    for key, val in panel_buffers.items():
+        config['detectors'][key]['buffer'] = val
+
+    # Convert to the selected tilt convention
```
Comment (joelvbernier) [inline hexrd/ui/utils.py:278]: This shouldn't be necessary, but let me check.

Comment (joelvbernier) [reply hexrd/ui/utils.py:278]: got it!

**Review by joelvbernier — APPROVED** [review]

**Comment by joelvbernier** [comment]
LGTM!

**Comment by psavery** [comment]
This fulfilled some of the requirements for hexrd/hexrd#239

**Pull request merged**